### PR TITLE
Item: improve keyboard interaction and semantics

### LINKED ISSUE #3714: [Enhancement] Item: improve keyboard interaction and semantics
- [x] remove the `tabindex` attribute from `ion-item`. Only selectable items should be focusable with the tab key, and this is already ensured by Ionic renderring a native html button inside the shadow dom when setting `<ion-item [button]="true" ...>`.
- [x] check for slotted `ion-toggle` in `_isIonicButton` logic (should NOT be a button in this case)

### PR DESCRIPTION
**Pull request by jakobe**
## Which issue does this PR close?

This PR closes #3714

## What is the new behavior?

Removes `tabindex` from `ion-item`, as the item should only be focusable if it renders a native button (`[selectable]="true"`) or contains an interactive control (checkbox, radio or toggle).

In both cases the nested controls are focusable out of the box.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Checklist:

The following tasks should be carried out in sequence in order to follow [the process of contributing](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#the-process-of-contributing) correctly.

### Reminders
- [x] Make sure you have implemented tests following the guidelines in: "[The good: Test](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Test)".
- [x] Make sure you have updated the cookbook with examples and showcases (for bug fixes, enhancements & new components).

### Review  
- [x] Determine if your changes are a fix, feature or breaking-change, and add the matching label to your PR. If it is tooling, dependency updates or similar, add ignore-for-release.
- [x] Do a [self-review](https://github.com/kirbydesign/designsystem/wiki/The-Good%3A-Self-review).
- [x] Request that the changes are code-reviewed 
- [ ] Request that the changes are [UX reviewed](https://github.com/kirbydesign/designsystem/blob/develop/.github/CONTRIBUTING.md#ux-review) (only necessary if your PR introduces visual changes)

When the pull request has been approved it will be merged to `develop` by Team Kirby.

### CODE REVIEW

File: apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.ts [hunk 1]
```diff
@@ -12,8 +12,8 @@ import {
 export class ItemShowcaseComponent {
   properties: ApiDescriptionProperty[] = [
     {
-      name: 'Selectable',
-      description: 'Make item selectable',
+      name: 'selectable',
+      description: 'If `true`, a native button tag will be rendered and the item will be tappable.',
```
Comment (RasmusKjeldgaard) [inline apps/cookbook/src/app/showcase/item-showcase/item-showcase.component.ts]: Consider adding a bit of context
```suggestion
      description: 'If `true`, a native button tag will be rendered under the hood and the item will become interactive.',
```

File: libs/designsystem/item/src/item.component.html [hunk 2]
```diff
@@ -1,9 +1,8 @@
 <ion-item
   lines="none"
   [attr.disabled]="disabled"
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/item/src/item.component.html:3]: Consider also handling this improvement in this PR, as it seems tightly coupled to the button-functionality.
```suggestion
  [disabled]="disabled"
```

File: libs/designsystem/item/src/item.component.integration.spec.ts [hunk 3]
```diff
@@ -36,28 +39,46 @@ describe('ItemComponent with LabelComponent', () => {
...
-    });
+        );
+        ionItem = spectator.queryHost('ion-label');
+        await TestHelper.whenReady(ionItem);
+        labelElements = spectator.queryAll(
+          'ion-item ion-label > :is(h1, h2, h3, h4, h5, h6, p, data)'
+        );
+      });
+
+      it('should render general header, data and paragraph elements with correct font-weight', () => {
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/item/src/item.component.integration.spec.ts]:
```suggestion
      it('should render heading, data and paragraph elements with correct font-weight', () => {
```

File: libs/designsystem/item/src/item.component.integration.spec.ts [hunk 4]
```diff
@@ -36,28 +39,46 @@ describe('ItemComponent with LabelComponent', () => {
...
+            expect(e).toHaveComputedStyle({ 'font-weight': fontWeight('normal') });
+          });
+      });
     });
+  });
+
+  describe('when configured with selectable="true" and nested interactive element', () => {
+    const nestedInteractiveElements = ['kirby-checkbox', 'kirby-radio', 'kirby-toggle'];
+    nestedInteractiveElements.forEach((element) => {
+      it(`should not render native button when has nested ${element}`, async () => {
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/item/src/item.component.integration.spec.ts]:
```suggestion
      it(`should not render native button when there is a nested ${element}`, async () => {
```

File: libs/designsystem/item/src/item.component.integration.spec.ts [hunk 5]
```diff
@@ -36,28 +39,46 @@ describe('ItemComponent with LabelComponent', () => {
...
+        labelElements
+          .filter((e) => !!e.attributes.getNamedItem('detail'))
+          .forEach((e) => {
+            expect(e).toHaveComputedStyle({ 'font-weight': fontWeight('normal') });
+          });
+      });
     });
+  });
+
+  describe('when configured with selectable="true" and nested interactive element', () => {
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/item/src/item.component.integration.spec.ts]: Consider removing this if you do add the suggestion below
```suggestion
  describe('when configured with selectable="true"', () => {
```

File: libs/designsystem/item/src/item.component.integration.spec.ts [hunk 6]
```diff
@@ -36,28 +39,46 @@ describe('ItemComponent with LabelComponent', () => {
...
+        await TestHelper.whenReady(ionItem);
 
-    it('should render detail data and paragraph elements with correct font-weight', () => {
-      labelElements
-        .filter((e) => !!e.attributes.getNamedItem('detail'))
-        .forEach((e) => {
-          expect(e).toHaveComputedStyle({ 'font-weight': fontWeight('normal') });
-        });
+        const nativePart = ionItem.shadowRoot.querySelector('[part="native"]');
+        expect(nativePart.tagName).not.toEqual('BUTTON');
```
Comment (RasmusKjeldgaard) [inline libs/designsystem/item/src/item.component.integration.spec.ts:80]: Nice 👌

**Review by RasmusKjeldgaard — COMMENTED** [review]
Really solid!

File: libs/designsystem/item/src/item.component.html (hunk 2 above)
Comment (jakobe) [reply libs/designsystem/item/src/item.component.html:3]: There are lot of other stuff related to the disabled which I would like to handle in a separate PR 🤞🏻

Comment (RasmusKjeldgaard) [reply libs/designsystem/item/src/item.component.html:3]: Good call!

**Review by RasmusKjeldgaard — DISMISSED** [review]

**Review by RasmusKjeldgaard — APPROVED** [review]

**Pull request merged**